src/routes/profile.js

```diff
@@ -155,47 +198,176 @@ router.get(
+      // TODO-kev: Function for 1MB? Enum object?
+      req.flash('error', 'File size exceeds the limit: 1MB.');
+      return res.redirect(
+        `/profile/${req.params.profileUsername}/changeavatar`,
+      );
+    } else if (err) {
+      console.log(err);
+    }
+    next();
+  });
+};
+
 // Update the customavatar
 router.post(
   '/:profileUsername/changeavatar',
   checkProfileOwnership,
-  (req, res) => {
-    console.log('Recieved change avatar');
-    console.log(`For user ${req.params.profileUsername}`);
-    console.log(`Res link: ${req.body.reslink}`);
-    console.log(`Spy link: ${req.body.spylink}`);
-    console.log(`Message to mod: ${req.body.msgToMod}`);
+  upload,
+  async (req, res) => {
+    const result = await validateUploadAvatarRequest(
+      req.params.profileUsername,
+      req.files,
+    );
 
-    // sometimes https links dont show up correctly
-    // req.body.reslink.replace("https", "http");
-    // req.body.spylink.replace("https", "http");
+    if (!result.valid) {
+      res.status(400).send(result.errMsg);
+      return;
+    }
 
-    const avatarRequestData = {
-      forUsername: req.params.profileUsername.toLowerCase(),
+    const msgToMod = req.body.msgToMod
+      ? sanitizeHtml(req.body.msgToMod)
+      : 'No message provided.';
 
-      resLink: sanitizeHtml(req.body.reslink),
-      spyLink: sanitizeHtml(req.body.spylink),
-      msgToMod: sanitizeHtml(req.body.msgToMod),
+    // Upload valid avatar requests to s3 bucket
+    const avatarRes = req.files['avatarRes'][0];
+    const avatarSpy = req.files['avatarSpy'][0];
 
-      dateRequested: new Date(),
+    const avatarLinks = await s3.uploadAvatarRequest(
+      req.params.profileUsername,
+      avatarRes.buffer,
+      avatarSpy.buffer,
+    );
 
+    const avatarRequestData = {
+      forUsername: req.params.profileUsername.toLowerCase(),
+      resLink: avatarLinks[0],
+      spyLink: avatarLinks[1],
+      msgToMod: msgToMod,
+      dateRequested: new Date(),
       processed: false,
     };
 
     avatarRequest.create(avatarRequestData, (err, createdRequest) => {
+      // TODO-kev: What happens here with the error? Send an error response?
       if (err) {
         console.log(err);
       } else {
-        req.flash(
-          'success',
-          'Your submission was received! Please wait for a moderator to process your request.',
-        );
-        res.redirect(`/profile/${req.params.profileUsername}`);
+        res
+          .status(200)
+          .send(
+            'Your submission was received! Please wait for a moderator to process your request.',
+          );
       }
     });
+
+    console.log(
+      `Received change avatar request for user: ${req.params.profileUsername} msgToMod: ${msgToMod} resLink: ${avatarLinks[0]} spyLink: ${avatarLinks[1]}`,
+    );
+
+    // TODO-kev: Why does webstorm say return is unnecessary but we need it to not set additional headers?
+    return;
```

Comment:
Try to remove this guy after you extract out the above callback into async/await. It might resolve this issue you're seeing.

Reply:
yup it did!